server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -2136,6 +2136,7 @@ public Host findUpAndEnabledHostWithAccessToStoragePools(List<Long> poolIds) {
         if (hostIds.isEmpty()) {
             return null;
         }
+        Collections.shuffle(hostIds);
```

Comment:
I think we could add some log to this method, about which host was selected (or not).

Reply:
There's logging elsewhere, where the returned value is used.

Reply:
I might be wrong on this one, but I think that there are not many logs regarding the chosen host to migrate.
Call hierarchy that I've taken a quick check does not present logs. Assuming your comment, this would be logged later on in the call hierarchy ( I haven't checked all the way). Even if it is, you can see that many exceptions can be thrown not too late after the host ID is retrieved.
```
> StorageManagerImpl.findUpAndEnabledHostWithAccessToStoragePools(List<Long>)  (com.cloud.storage)
    > VolumeServiceImpl.copyManagedVolume(VolumeInfo, DataStore)  (org.apache.cloudstack.storage.volume)
        > VolumeServiceImpl.copyVolume(VolumeInfo, DataStore)  (org.apache.cloudstack.storage.volume)
            > VolumeOrchestrator.copyVolumeFromSecToPrimary(VolumeInfo, VirtualMachine, VirtualMachineTemplate, DataCenter, Pod, Long, ServiceOffering, ...)  (org.apache.cloudstack.engine.orchestration)
            ...
            > VolumeOrchestrator.migrateVolume(Volume, StoragePool)  (org.apache.cloudstack.engine.orchestration)
            ...
            > VolumeApiServiceImpl.orchestrateExtractVolume(long, long)  (com.cloud.storage)
            ...
```

Additionally, the `canHostAccessStoragePools` called inside the `findUpAndEnabledHostWithAccessToStoragePools` can return false. If the host cannot access, not much is logged as well.

Please, let us know in case we are missing something here.

Reply:
The logging is not of the randomised list but the selected host is logged by the deployment planner and startvm cmd.